portals/scripts/setup.ps1

```
@@ -0,0 +1,668 @@
+    } finally {
+        $ErrorActionPreference = $prev
+    }
+    return $ok
+}
+
+# Verify the tools this script depends on before doing any work. openssl is a
+# hard requirement for every path (cert, keys, and password generation); the
+# bcrypt tool (htpasswd or docker) is only needed when a new admin credential
+# is generated, so its absence is a warning here and a hard failure later in
+# Get-BcryptHash if and when hashing is actually attempted.
+function Test-Prerequisites {
```

PSScriptAnalyzer (1.25.0)

[warning] The cmdlet 'Test-Prerequisites' uses a plural noun. A singular noun should be used instead.

Suggested fix: Singularized correction of 'Test-Prerequisites'

(PSUseSingularNouns)